spec/main.tex

```diff
@@ -1123,28 +1141,41 @@ \subsubsection{Threshold ($t$-out-of-$n$) Presinging}
 
 \begin{description}
 
-\item[\bf Input.] Size of signing set $t$, identities of parties in signing set,
- index $i\in [t]$, index map\footnote{$S(i)$ is the index that $P_i$ had at key-generation time.} $S : [t] \to [n]$, key share $K_{S(i)}$, context separation string $\sid$, security level $L$, curve $\E$ with generator $G$.
+\item[\bf Input.] 
+  Size of signing set $t$,
+  identities of parties in signing set,
+  index $i\in [t]$,
+  index map\footnote{$S(i)$ is the index that $P_i$ had at key-generation time.} $S : [t] \to [n]$,
+  key share $K_{S(i)}$,
+  context separation string $\sid$,
+  security level $L$, curve $\E$ with generator $G$.
+
+  {\bf HD-wallets inputs:} additive $\text{shift} \in \Fq$ \footnote{Deriving additive shift is up to specific standard of HD-wallets, e.g. see \cite{bip32} or \cite{slip10}} ($\text{shift} = 0$ disables HD derivation)
 
     \item[\bf Setup.] The key share $K_{S(i)}$ contains $\threshold$, number of key holders~$n$, secret share $x'_{S(i)}$, parties' public shares $\vec X' = (X'_j)_{j \in [n]}$, a map $I : [n] \to \Fq \setminus \{0\}$, Paillier secret key $\sk_{S(i)}$, parties' Paillier keys $\vec N' = (N'_j)_{j \in [n]}$, and parties' auxiliary information $\vec R' = (s_j, t_j, \hat N)_{j \in [n]}$.
 
     \item[\bf Step 1.] Set $\sk_i = \sk_{S(i)}$ and 
-            $\vec R = (R'_{S(j)})_{j \in [n]}$. Then:
+                       $\vec R = (R'_{S(j)})_{j \in [n]}$. Then:
     \begin{itemize}
-           \item If shares are additive\footnote{In this case we have $t=n$.} shares of the private key, 
+        \item If shares are additive\footnote{In this case we have $t=n$.} shares of the private key, 
             set $x_{i} = x'_{S(i)}$, $\vec X = (X'_{S(j)})_{j \in [n]}$.
-
         
         \item If shares are Shamir secret shares of the private key: 
-
         \begin{itemize}
-        \item For $j \in [t]$, compute Lagrange coefficient $\lambda_j = \prod\limits_{m \in [t]\setminus\{j\}} \frac{I(S(m))}{I(S(m)) - I(S(j))} \bmod{q}$.
+            \item For $j \in [t]$, compute Lagrange coefficient $\lambda_j = \prod\limits_{m \in [t]\setminus\{j\}} \frac{I(S(m))}{I(S(m)) - I(S(j))} \bmod{q}$.
             \item Compute $x_{i} = \lambda_i \cdot x'_{S(i)}$.
             \item For $j \in [n]$, compute $X_j= \lambda_j \cdot X'_{S(j)}$; then set $\vec X = \{X_j\}_{j \in [t]}$.
         \end{itemize}
     \end{itemize}
 
-    \item[\bf Step 2.] Call the non-threshold presigning protocol from the previous section using inputs $t, i, x_i, \vec X$, $\sk_i$, $\vec R, \sid, L, \E$.
+    \item[\bf Step 2.] {\bf (HD-wallets)} If HD-wallets support enabled:
+    \begin{itemize}
+      \item Update $X_0 := X_0 + \text{shift} \cdot G$
```

Comment:
Might be read that we update the stored `X_0` in the keyshare, when we just use it locally and discard

Reply:
Changed wording

Reply:
Mmmm reads exactly the same to me. Maybe add a note "for this protocol invocation only"?

Reply:
I don't like it. The setup step states that key share contains secret share $x'_{S(i)}$ and list of public shares $\vec X'$. It's different from $x$ and $X$ that we define later. Why would someone assume that changing something derived from the key share must change the key share?

Reply:
$x$ and $X$ are completely different from the ones stored in the key share, at least because they are multiplied by lagrange coefficient (if it's threshold share). Would the reader assume that key share also needs to be multiplied at lagrnage coef?

Reply:
Ah you're right, then it made sense even before. Sorry for bothering

Reply:
No worries! "Update" does sound a bit misleading, changing it to "set" is probably for the best